packages/wonder-blocks-timing/src/hooks/__tests__/use-interval.test.ts

```diff
@@ -1,123 +1,520 @@
+            );
+
+            // Act
+            const isSet = result.current.isSet;
+
+            // Assert
+            expect(isSet).toBeFalsy();
+        });
+
+        it("is true when the interval is active", () => {
+            // Arrange
+            const {result} = renderHook(() => useInterval(() => {}, 1000));
+            act(() => {
+                result.current.set();
+            });
+
+            // Act
+            const isSet = result.current.isSet;
+
+            // Assert
+            expect(isSet).toBeTruthy();
+        });
+
+        it("is false when the interval is cleared", () => {
+            // Arrange
+            const {result} = renderHook(() => useInterval(() => {}, 1000));
+            act(() => {
+                result.current.set();
+                result.current.clear();
+            });
+
+            // Act
+            const isSet = result.current.isSet;
+
+            // Assert
+            expect(isSet).toBeFalsy();
+        });
+    });
+
+    describe("#set", () => {
+        it("should call setInterval", () => {
+            // Arrange
+            const intervalSpy = jest.spyOn(global, "setInterval");
+            const {result} = renderHook(() =>
+                useInterval(() => {}, 500, {
+                    schedulePolicy: SchedulePolicy.OnDemand,
+                }),
+            );
+
+            // Act
+            act(() => {
+                result.current.set();
+            });
+
+            // Assert
+            expect(intervalSpy).toHaveBeenNthCalledWith(
+                1,
+                expect.any(Function),
+                500,
+            );
+        });
+
+        it("should invoke setInterval to call the given action", () => {
+            // Arrange
+            const intervalSpy = jest.spyOn(global, "setInterval");
+            const action = jest.fn();
+            const {result} = renderHook(() =>
+                useInterval(action, 500, {
+                    schedulePolicy: SchedulePolicy.OnDemand,
+                }),
+            );
+
+            act(() => {
+                result.current.set();
+            });
+            const scheduledAction = intervalSpy.mock.calls[0][0];
+
+            // Act
+            scheduledAction();
+
+            // Assert
+            expect(action).toHaveBeenCalledTimes(1);
+        });
+
+        it("should clear the active interval", () => {
+            // Arrange
+            const action = jest.fn();
+            const {result} = renderHook(() =>
+                useInterval(action, 500, {
+                    schedulePolicy: SchedulePolicy.OnDemand,
+                }),
+            );
+            act(() => {
+                result.current.set();
+            });
+
+            // Act
+            act(() => {
+                result.current.set();
+                jest.advanceTimersByTime(501);
+            });
+
+            // Assert
+            expect(action).toHaveBeenCalledTimes(1);
+        });
+
```

Comment:
Given the title of the tests I would expect it to spy on `clearInterval` and assert that it was called instead of the action.